### PR TITLE
Add interop tests for grpc-dotnet repository

### PR DESCRIPTION
**Pull request by jtattermusch**
Interop tests against Grpc.AspNetCore.Server.

Inspired by https://github.com/grpc/grpc/pull/14120.

### CODE REVIEW

File: tools/run_tests/run_interop_tests.py [hunk 1]
```diff
@@ -173,6 +174,36 @@ def __str__(self):
...
+    def __init__(self):
+        self.server_cwd = '../grpc-dotnet/testassets/InteropTestsWebsite/bin/Debug/netcoreapp3.0'
+        self.safename = str(self)
+
+    def cloud_to_prod_env(self):
+        return {}
+
+    def client_cmd(self, args):
+        # client not supported
+        return ['dotnet', 'exec', 'InteropClient.cll'] + args
```
Comment (apolcyn) [inline tools/run_tests/run_interop_tests.py]: nit: `return ['echo', 'not supported']` ? `InteropClient.cll` isn't actually a client, right?

File: tools/dockerfile/interoptest/grpc_interop_aspnetcore/build_interop.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+git clone /var/local/jenkins/grpc-dotnet /var/local/git/grpc-dotnet
+
+# copy service account keys if available
+cp -r /var/local/jenkins/service_account $HOME || true
+
+cd /var/local/git/grpc-dotnet
+./build/get-grpc.sh
+
+cd testassets/InteropTestsWebsite
+dotnet build --configuration Debug
```
Comment (apolcyn) [inline tools/dockerfile/interoptest/grpc_interop_aspnetcore/build_interop.sh]: consider building the interop test in release mode? Realize that languages are mixed on this, though.

**Comment by jtattermusch** [comment]
https://github.com/grpc/grpc-dotnet/pull/93 needs to be merged first.

File: tools/dockerfile/interoptest/grpc_interop_aspnetcore/build_interop.sh (hunk 2 above)
Comment (jtattermusch) [reply tools/dockerfile/interoptest/grpc_interop_aspnetcore/build_interop.sh]: Currently other C# interop clients and servers are also built in dbg, so I think I will stick with that for consistency https://github.com/grpc/grpc/blob/aa207933cb5f62b4726bbd52bd109f9e875a8ec0/templates/tools/dockerfile/csharp_build_interop.sh.include#L32.

File: tools/run_tests/run_interop_tests.py (hunk 1 above)
Comment (jtattermusch) [reply tools/run_tests/run_interop_tests.py]: I will replace with something that actually fails - silently passing unimplemented test cases are bad.

**Comment by jtattermusch** [comment]
This is progress towards https://github.com/grpc/grpc-dotnet/issues/32

**Comment by jtattermusch** [comment]
Known failures: https://github.com/grpc/grpc/issues/17550

**Pull request merged**